resource-managers/mesos/src/test/scala/org/apache/spark/scheduler/cluster/mesos/MesosFineGrainedSchedulerBackendSuite.scala

```diff
@@ -80,9 +81,9 @@ class MesosFineGrainedSchedulerBackendSuite
   }
 
   test("Use configured mesosExecutor.cores for ExecutorInfo") {
-    val mesosExecutorCores = 3
+    val mesosExecutorCores = 3.0
```

Comment:
why is this change needed?

Reply:
The type of EXECUTOR_CORES is actually `double` instead of `int` or `long` and if my memory is right compiler complained. If we don't want to touch this we may need to change below line to `conf.set(EXECUTOR_CORES.key, mesosExecutorCores.toString)`

Reply:
The type of a number of cores is logically an integer and can be used where a double is expected. In a few cases a cast may be needed but doesn't require thinking of number of cores as a float

Reply:
I'm honestly not a expert on this but the documentation of `spark.mesos.mesosExecutor.cores` clarify that it can receive floating point number so the type still makes sense.

https://github.com/apache/spark/blob/master/docs/running-on-mesos.md

The default value is also explained as `1.0` instead of 1. I'll update the config entry to reflect the sync.

Reply:
Oh right mesos can specify fractional cores. OK this makes sense then.